apis/v1beta1/config.go

```diff
@@ -573,19 +596,43 @@ func AddPrometheusMetricsEndpoint(host string, port int32) otelConfig.MetricRead
 
 // GetTelemetry serves as a helper function to access the fields we care about in the underlying telemetry struct.
 // This exists to avoid needing to worry extra fields in the telemetry struct.
-func (s *Service) GetTelemetry() *Telemetry {
+func (s *Service) GetTelemetry(logger *logr.Logger) *Telemetry {
 	if s.Telemetry == nil {
+		if logger != nil {
+			logger.V(1).Info("no telemetry configuration found")
+		}
 		return nil
 	}
+
 	// Convert map to JSON bytes
 	jsonData, err := json.Marshal(s.Telemetry)
 	if err != nil {
+		if logger != nil {
+			logger.Error(err, "failed to marshal telemetry configuration to JSON", "telemetry", s.Telemetry.Object)
+		}
 		return nil
 	}
+
+	if logger != nil {
+		logger.V(2).Info("marshaled telemetry configuration", "json", string(jsonData))
+	}
+
 	t := &Telemetry{}
 	// Unmarshal JSON into the provided struct
 	if err := json.Unmarshal(jsonData, t); err != nil {
+		if logger != nil {
+			logger.Error(err, "failed to unmarshal telemetry configuration - this may indicate invalid configuration",
+				"json", string(jsonData), "originalConfig", s.Telemetry.Object)
+		}
 		return nil
 	}
+
+	if logger != nil {
```

Comment:
You are doing this because of the tests, right? Maybe you can just use https://pkg.go.dev/go.uber.org/zap#NewNop instead of providing `nil` as parameter.

Reply:
Sure, let me do it.